test/create-data-stream.js

```diff
@@ -0,0 +1,191 @@
+          t.ok(feature.properties.icon)
+          t.same(feature.properties.type, batch[0].value.tags.type)
+          t.same(feature.properties.area, batch[0].value.tags.area)
+          t.same(feature.properties['animal-type'], batch[0].value.tags['animal-type'])
+          done()
+        }))
+      })
+    })
+  })
+})
+
+test('createDataStream: filter only point', (t) => {
+  var mapeo = helpers.createApi()
+  mapeo.on('error', console.error)
+
+  function done () {
+    t.end()
+    mapeo.close()
+  }
+
+  var filter = ['==', '$type', 'node']
+  var batchAll = data.polygon.batch.concat(data.way.batch, data.node.batch)
+
+  mapeo.osm.ready(function () {
+    mapeo.osm.batch(batchAll, (err) => {
+      t.error(err)
+      var rs = mapeo.createDataStream({ filter })
+      rs.pipe(concat((geojson) => {
+        var actual = JSON.parse(geojson)
+        actual.features = actual.features.map((f) => {
+          delete f.id
+          return f
+        })
+        t.same(actual, data.node.expected)
+        done()
+      }))
+    })
+  })
+})
+
+test('createDataStream: filter only with tag \'interesting\'', (t) => {
+  var mapeo = helpers.createApi()
+  mapeo.on('error', console.error)
+
+  function done () {
+    t.end()
+    mapeo.close()
+  }
+
+  var filter = ['has', 'interesting']
+  var batchAll = data.polygon.batch.concat(data.way.batch, data.node.batch)
+  var expected = {
+    type: 'FeatureCollection',
+    // TODO: Order is not guaranteed!
+    features: [...data.way.expected.features, ...data.node.expected.features]
+  }
+
+  mapeo.osm.ready(function () {
+    mapeo.osm.batch(batchAll, (err) => {
+      t.error(err)
+      var rs = mapeo.createDataStream({ filter })
+      rs.pipe(concat((geojson) => {
+        var actual = JSON.parse(geojson)
+        actual.features = actual.features.map((f) => {
+          delete f.id
+          return f
+        })
+        t.same(actual, expected)
+        done()
+      }))
+    })
+  })
+})
+
+test('createDataStream: filter only with tag \'area\' == \'yes\'', (t) => {
+  var mapeo = helpers.createApi()
+  mapeo.on('error', console.error)
+
+  function done () {
+    t.end()
+    mapeo.close()
+  }
+
+  var filter = ['==', 'area', 'yes']
+  var batchAll = data.polygon.batch.concat(data.way.batch, data.node.batch)
+
+  mapeo.osm.ready(function () {
+    mapeo.osm.batch(batchAll, (err) => {
+      t.error(err)
+      var rs = mapeo.createDataStream({ filter })
+      rs.pipe(concat((geojson) => {
+        var actual = JSON.parse(geojson)
+        actual.features = actual.features.map((f) => {
+          delete f.id
+          return f
+        })
+        t.same(actual, data.polygon.expected)
+        done()
+      }))
+    })
+  })
+})
+
+// **TODO**: Add test for sync file creation
+
+function getOsmStr (mapeo, cb) {
+  var query = mapeo.osm.query([-Infinity, -Infinity, Infinity, Infinity])
+  query.pipe(concat((data) => {
+    cb(null, data)
+  })).on('error', cb)
```

Comment:
Unlikely that it'll be a problem, but `pump` will catch all pipeline errors, just in case `query` errors for some reason.

Reply:
Yeah. This is actually from the original test code. There are a few things like this in Mapeo Core tests that need cleaned up at some point.